base/iostream.jl

```diff
@@ -485,11 +485,13 @@ function read(s::IOStream)
         if pos > 0
             sz -= pos
         end
-    catch
+    catch err
+        err isa IOError || rethrow(err)
```

Comment:
I suspect this was also intended to catch MethodErrors for streams where `filesize` doesn't work. I guess it would be better to have a `sizeestimate` function that can default to `0` for all streams.

Reply:
I guess I should just delete this `try/catch` then, since this doesn't appear to particularly likely ever to fail. The only case appears to be:
```
     [EIO]              An I/O error occurs while reading from or writing to the file system.
```

Reply:
OK, that was correct for any posix systems, but not for libuv on Win32. On there, we can also get EINVAL errors since their `stat` function only works on some types of IO handles.

There can't be any MethodErrors, since the call tree here is all statically decided.